src/core/term_dag.rs

```diff
@@ -164,6 +164,78 @@ impl<Witness, App: Application> TermDag<Witness, App> {
         Rc::new(TermDag::Jet(jet))
     }
 
+    /// Create a DAG that takes any input and returns `value` as constant output.
+    pub fn scribe(value: &Value) -> Rc<TermDag<Witness, App>> {
+        match value {
+            Value::Unit => TermDag::unit(),
+            Value::SumL(l) => {
+                let l = TermDag::scribe(l);
+                TermDag::injl(l)
+            }
+            Value::SumR(r) => {
+                let r = TermDag::scribe(r);
+                TermDag::injr(r)
+            }
+            Value::Prod(l, r) => {
+                let l = TermDag::scribe(l);
+                let r = TermDag::scribe(r);
+                TermDag::pair(l, r)
+            }
+        }
+    }
+
+    /// Create a DAG that takes any input and returns bit `0` as constant output.
+    pub fn bit_false() -> Rc<Self> {
+        TermDag::injl(TermDag::unit())
+    }
+
+    /// Create a DAG that takes any input and returns bit `1` as constant output.
+    pub fn bit_true() -> Rc<Self> {
+        TermDag::injr(TermDag::unit())
+    }
+
+    /// Create a DAG that takes a bit and an input,
+    /// such that the `left` child is evaluated on the input if the bit is `1` _(if branch)_
+    /// and the `right` child is evaluated on the input otherwise _(else branch)_.
+    ///
+    /// _The input and output types of both children must be the same._
```

Comment:
In 32dcdfc863406a71fc46022f5adb43c242f7f973:

All the "must be"s in this commit should have corresponding assertions in the code (or maybe these methods should just return `Result`s).

Reply:
Assertions would be helpful, but unfortunately `TermDag` contains no type information. We could convert the children into linear programs and do type checking every time we construct a larger DAG, but this seems like a bit much.

I added the "type hints" to the documentation to help resolve type checking issues. I will update the way I wrote it, but for now it will be limited to documentation only.

Reply:
Ah! I understand. Maybe you could add to the comment something like "if this is not upheld, type inference will fail, possibly in surprising and nonlocal ways".

Reply:
Type checking is a huge source of frustration when writing programs from scratch. I think it would be easy to write a `TypedDag` that includes type information and can check its children as it is being constructed. We could convert that straight to `TypedProgram`, making use of the generic DAG traversal code. It's an idea for a next PR.